### PR TITLE
MAINT: Bump to 3.9 and NumPy 2

### LINKED ISSUE #65: New release for numpy 2.0 necessary?
Hi,

Is it possible we need a new release of pymeshfix now that numpy 2.0.0 is out? Or update dependency requirements in `setup.py` so that only numpy 1.X is supported?

I get the following error during unit tests of a library that uses `pymeshfix`

> lib\my-lib\dev\meshfix.py:9: in <module>
>     from pymeshfix import _meshfix
> .venv\lib\site-packages\pymeshfix\__init__.py:2: in <module>
>     from pymeshfix import _meshfix  # noqa: F401
> pymeshfix\\cython\\_meshfix.pyx:1: in init pymeshfix._meshfix
>     ???
> E   ImportError: numpy.core.multiarray failed to import (auto-generated because you didn't call 'numpy.import_array()' after cimporting numpy; use '<void>numpy._import_array' to disable if you are certain you don't need it).
> ------------------------------- Captured stderr -------------------------------
> 
> A module that was compiled using NumPy 1.x cannot be run in
> NumPy 2.0.0 as it may crash. To support both 1.x and 2.x
> versions of NumPy, modules must be compiled with NumPy 2.0.
> Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.
> 
> If you are a user of the module, the easiest solution will be to
> downgrade to 'numpy<2' or try to upgrade the affected module.
> We expect that some modules will need time to support NumPy 2.
>

### PR DESCRIPTION
**Pull request by larsoner**
1. Make 3.9 the min Python. Should be safe enough nowadays since NumPy 2.0 didn't release 3.8 wheels.
2. Build against NumPy 2.0
3. Set version in prep for release

@akaszynski I suggest we merge this PR then cut a 0.17.0 release

Closes #65

### CODE REVIEW

File: pymeshfix/_version.py
Comment (akaszynski) [inline pymeshfix/_version.py:10]:
```suggestion
version_info = 0, 18, dev0
```

Comment (akaszynski) [reply pymeshfix/_version.py:10]: Standard practice for pyvista repos is to keep main branch with the next minor ahead `dev0` (like numpy).

**Comment by akaszynski** [comment]
> @akaszynski I suggest we merge this PR then cut a 0.17.0 release

Sure! After merging this, create a new branch `release/0.17` and then bump the version on that branch to `0.17.0`. Push that commit and then tag with `v0.17.0`. CI should take care of the rest.

**Review by akaszynski — DISMISSED** [review]
LGTM (see https://github.com/pyvista/pymeshfix/pull/66#discussion_r1647666432).

File: pymeshfix/_version.py
Comment (akaszynski) [inline pymeshfix/_version.py:10]:
```suggestion
version_info = 0, 18, "dev0"
```

**Comment by larsoner** [comment]
Pushed, feel free to approve and/or merge @akaszynski (I can't merge myself without an approving review, but marked for merge-when-green-and-approved)

**Review by akaszynski — APPROVED** [review]
Reapproving.

**Comment by akaszynski** [comment]
Seems that we need to fix our required CI checks...

**Comment by larsoner** [comment]
Also I don't see the point of building both `universal2` and `x86_64` wheels on macOS. To me it makes the most sense to build x86_64 on macos-13 and arm64 on macos-14. I'll open a PR to update to this (I think it's cibuildwheel suggested/best practice nowadays)

**Pull request merged**
